### DISCUSSION TITLE
How to load terrain data in webp format using deck.gl

### DISCUSSION THREAD

**Discussion by Paweled** [visgl/deck.gl#6070] [General]
How to load terrain data in webp format using deck.gl

**Comment by Pessimistress** [comment]
It's not possible right now without a hack. But it should be a pretty simple feature to add.

@chrisgervang @kylebarron it seems that the TerrainLoader just uses ImageLoader to parse the incoming image, so there's no real reason to restrict the format to png? We can use the `mimeType` trick discussed here: https://github.com/visgl/deck.gl/pull/5934

**Reply by kylebarron** [reply]
I agree there's no reason to restrict to PNG. Right now it wouldn't work because the loader restricts to png [here](https://github.com/visgl/loaders.gl/blob/8f79b48d87aa05d86a9bbdddbf8096f7e7efaf5a/modules/terrain/src/terrain-loader.ts#L13-L14)? We could add more image formats to that list?

**Reply by Pessimistress** [reply]
I believe we can just add `mimeType: 'application/x.terrain'` to the loader options in TerrainLayer's getTileData.

**Reply by warflash** [reply]
Hey, just stumbled across this discussion.
Couldn't find any issues regarding other image formats so I'm wondering if this was closed elsewhere?
If it's not added yet I'd appreciate the addition as webp would shave of quite a bit of size for height data

**Comment by ibgreen** [comment]
@warflash If you want to use the same actual encoding, just stored in webp instead of jpeg, you might be able to just modify the `extensions` and/or `mimeTypes` field of the `TerrainLoader` after you import it.

https://github.com/visgl/loaders.gl/blob/master/modules/terrain/src/terrain-loader.ts#L13

**Reply by warflash** [reply]
Alright, I'll give that a try then👍 
Was trying to avoid modifying loaders directly since I couldn't find an example in the docs on how to do so yet, but will take another look!
Thanks for your response

**Reply by Pessimistress** [reply]
I'm working on a proper fix of this in the TerrainLayer.

**Reply by Pessimistress** [reply]
https://github.com/visgl/loaders.gl/pull/2040
